tests/tests/connectors/test-class-connector-gravityforms.php

```diff
@@ -0,0 +1,949 @@
+				'fields' => $this->create_fields( $fields )
+			)
+		);
+		$form_id = \GFAPI::add_form( $form );
+
+		return $form_id;
+	}
+
+	public function create_fields( $fields = array() ) {
+		$fields = ! empty( $fields ) ? $fields : array(
+			array(
+				'type' => 'text',
+				'id' => 1,
+				'label' => 'Single Line Text',
+				'adminLabel' => '',
+				'isRequired' => false,
+				'size' => 'medium',
+				'errorMessage' => '',
+				'visibility' => 'visible',
+				'inputs' => null,
+				'formId' => 2,
+				'description' => 'I am a single line text field.',
+				'allowsPrepopulate' => false,
+				'inputMask' => false,
+				'inputMaskValue' => '',
+				'inputMaskIsCustom' => false,
+				'maxLength' => '',
+				'inputType' => '',
+				'labelPlacement' => '',
+				'descriptionPlacement' => '',
+				'subLabelPlacement' => '',
+				'placeholder' => '',
+				'cssClass' => '',
+				'inputName' => '',
+				'noDuplicates' => false,
+				'defaultValue' => '',
+				'choices' => '',
+				'productField' => '',
+				'enablePasswordInput' => '',
+				'multipleFiles' => false,
+				'maxFiles' => '',
+				'calculationFormula' => '',
+				'calculationRounding' => '',
+				'enableCalculation' => '',
+				'disableQuantity' => false,
+				'displayAllCategories' => false,
+				'useRichTextEditor' => false,
+				'checkboxLabel' => '',
+				'pageNumber' => 1,
+				'fields' => '',
+				'displayOnly' => '',
+			),
+		);
+
+		return $fields;
+	}
+
+	public function test_callback_gform_after_save_form() {
+		$this->mock->expects( $this->atLeastOnce() )
+			->method( 'log' )
+			->with(
+				$this->equalTo( __( '"%1$s" form %2$s', 'stream' ) ),
+				$this->equalTo(
+					array(
+						'title'  => 'Test form',
+						'action' => 'created',
+					)
+				),
+				$this->greaterThan( 0 ),
+				$this->equalTo( 'forms' ),
+				$this->equalTo( 'created' )
+			);
+
+		// Do stuff.
+		$form_id = $this->create_form();
+
+		$form_meta = \GFFormsModel::get_form_meta( $form_id );
+		do_action( 'gform_after_save_form', $form_meta, true, array() );
+
+		$this->assertFalse( 0 === did_action( 'wp_stream_test_callback_gform_after_save_form' ) );
+	}
+
+	public function test_callback_gform_pre_confirmation_save() {
+		$this->mock->expects( $this->atLeastOnce() )
+			->method( 'log' )
+			->with(
+				$this->equalTo( __( '"%1$s" confirmation %2$s for "%3$s"', 'stream' ) ),
+				$this->equalTo(
+					array(
+						'title'      => 'Test confirmation',
+						'action'     => 'created',
+						'form_title' => 'Test form',
+					)
+				),
+				$this->greaterThan( 0 ),
+				$this->equalTo( 'forms' ),
+				$this->equalTo( 'updated' )
+			);
+
+		// Do stuff.
+		$confirmation = array(
+			'id'          => uniqid(),
+			'name'        => 'Test confirmation',
+			'isDefault'   => true,
+			'type'        => 'message',
+			'message'     => 'Lorem ipsum dolor...',
+			'url'         => '',
+			'pageId'      => '',
+			'queryString' => '',
+		);
+
+		$form_id      = $this->create_form(
```

Comment:
Align spacing.

Reply:
@kidunot89 realizing this is aligned with the $confirmation variable. Looks a bit weird. I will allow you to decide if it should stay.